supervisor/daemon/wireguard/monitor.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package wireguard
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/mysteriumnetwork/node/supervisor/daemon/wireguard/wginterface"
+)
+
+// Monitor creates/deletes the WireGuard interfaces and keeps track of them.
+type Monitor struct {
+	interfaces map[string]*wginterface.WgInterface
+	sync.Mutex
```

Comment:
If you are embedding mutex to the public struct, it can be locked from outside by calling `.Lock()`. This will break internal logic.
```suggestion
	mu sync.Mutex
```